gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/cypher/capf/result/CAPFQueryResult.java

```diff
@@ -201,7 +201,7 @@ public GraphCollection getGraphs() {
       .map(new CreateGraphHeadWithProperties(
         entityFieldsCount,
         entityFieldsCount + otherFieldsCount,
-        config.getGraphHeadFactory(),
+        config.getLogicalGraphFactory().getGraphHeadFactory(),
```

Comment:
You can replace the `config` attribute with an attribute of type `GraphCollectionFactory`. As a result, there is no need for the config anymore in this class.